packages/util/src/units.ts

```diff
@@ -0,0 +1,118 @@
+// Source of constants: https://www.languagesandnumbers.com/articles/en/ethereum-ether-units/
+export enum Unit {
+  Wei = 'wei',
+  Kwei = 'kwei',
+  Mwei = 'mwei',
+  Gwei = 'gwei',
+  Micro = 'micro',
+  Milli = 'milli',
+  Ether = 'ether',
+  Kether = 'kether',
+  Megaether = 'megaether',
+  Gether = 'gether',
+}
+
+export enum UnitValue {
+  'wei' = -18,
+  'kwei' = -15,
+  'mwei' = -12,
+  'gwei' = -9,
+  'micro' = -6,
+  'milli' = -3,
+  'ether' = 1,
+  'kether' = 3,
+  'megaether' = 6,
```

Comment:
Who is using these units? And why `megather` and not `Mether`? Why not have `nanoether`/`nanoeth` which is used in some EIPs (like 1559)? How about the other old/odd units (`finney`/`shannon`/etc)?

The library this tries to replace has a lot more units: https://github.com/ethereumjs/ethereumjs-units/blob/master/units.json

I am not sure it is a good idea to support all these units, but I'd probably suggest to have at least two functions, a strict and a loose one. The strict should only support `wei`/`gwei`/`eth` and maybe `nanoeth`. The loose one can support all the legacy ones.

Reply:
Thanks for the additional context, we need to bring all this together with the overall API design, so we should give this some thought.

ATM (here) we are on a path where we have convenience functions for the most common unit conversions (so e.g. `fromGwei()`, which are wrapping the generic `convert()` function.

I would be a fan of instead of splitting this up in strict and loose but rather throwing obscure ones out and eventually take one or two missed in.

So that would mean:

- kether, mether, gether, tether out (would be in strong support, totally obscure and not suited given the value of ETH)
- For the lower ETH ones I guess we just might want to keep everything in for completeness + who knows what denominations become useful due to price evolution in some areas (gas for L2s, whatever)
- nanoeth == gwei, right? So we could add an alias here for the convenience functions, not really sure if this is necessary though (but also no strong opinion)

Generally: my "unit usage" feeling is limited, so other voices and additions here very welcome.

Reply:
> Who is using these units? And why megather and not Mether? Why not have nanoether/nanoeth which is used in some EIPs (like 1559)? How about the other old/odd units (finney/shannon/etc)? 

To be honest, I searched for Ethereum units on the internet and got these, I linked the URL in the code as a comment. I will update to the ones you suggested.

As Holger said:
> ATM (here) we are on a path where we have convenience functions for the most common unit conversions (so e.g. fromGwei(), which are wrapping the generic convert() function.

I was thinking that probably creating this `to${Unit}` methods it's a way to do these strict functions. I thought `Units.toGwei(10)` can be a good interface to make code easier to read

In advance, I want to say thank you for taking the time to review this :-)